pkg/provider/aws/aws.go

```diff
@@ -0,0 +1,149 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"emperror.dev/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/ssm"
+
+	"github.com/bank-vaults/secret-init/pkg/provider"
+)
+
+var ProviderName = "aws"
+
+type Provider struct {
+	sm  *secretsmanager.SecretsManager
+	ssm *ssm.SSM
+}
+
+func NewProvider(config *Config) *Provider {
+	return &Provider{
+		sm:  secretsmanager.New(config.session),
+		ssm: ssm.New(config.session),
+	}
+}
+
+func (p *Provider) LoadSecrets(_ context.Context, paths []string) ([]provider.Secret, error) {
+	var secrets []provider.Secret
+
+	for _, path := range paths {
+		split := strings.SplitN(path, "=", 2)
+		originalKey, secretID := split[0], split[1]
+
+		// valid secretsmanager secret examples:
+		// arn:aws:secretsmanager:region:account-id:secret:secret-name
+		// secretsmanager:secret-name
+		if strings.Contains(secretID, "secretsmanager:") {
+			secret, err := p.sm.GetSecretValue(
+				&secretsmanager.GetSecretValueInput{
+					SecretId: aws.String(secretID),
+				})
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get secret from AWS secrets manager")
+			}
+
+			if json.Valid([]byte(aws.StringValue(secret.SecretString))) {
+				var secretValue map[string]interface{}
+				err := json.Unmarshal([]byte(aws.StringValue(secret.SecretString)), &secretValue)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to unmarshal secret value")
+				}
+
+				// If there is only one key in the secret, append it directly
+				if len(secretValue) == 1 {
+					for _, value := range secretValue {
+						secretToAppend, err := appendSecret(originalKey, value)
+						if err != nil {
+							return nil, errors.Wrap(err, "failed to append secret")
+						}
+
+						secrets = append(secrets, secretToAppend)
+					}
+
+					continue
+				}
+
+				// If the secret is a JSON object, append it as a single secret
+				JSONValue, err := json.Marshal(secretValue)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to marshal secret value")
+				}
+
+				secretToAppend, err := appendSecret(originalKey, JSONValue)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to append secret")
+				}
+
+				secrets = append(secrets, secretToAppend)
+			} else {
+				// If the secret is not a JSON object, append it directly
+				secretToAppend, err := appendSecret(originalKey, aws.StringValue(secret.SecretString))
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to append secret")
+				}
+
+				secrets = append(secrets, secretToAppend)
+			}
```

Comment:
this can also be moved into a helper function:
```go
func parseSecretValueFromSM(secret []byte) ([]byte, error) {
	// Return the secret if it doesnt contain valid JSON
	if !json.Valid(secret) {
		return secret, nil
	}

	// Unmarshal secret into map to load secret's key-value data
	var secretData map[string]interface{}
	if err := json.Unmarshal(secret, &secretData); err != nil {
		return nil, fmt.Errorf("failed to unmarshal secret: %w", err)
	}

	// If the map contains a single KV, the actual secret is the value
	if len(secretData) == 1 {
		for _, value := range secretData {
			value, err := json.Marshal(value)
			if err != nil {
				return nil, fmt.Errorf("failed to marshal secret from map: %w", err)
			}
			return value, nil
		}
	}

	// Secret contains multiple sub-secrets, return all unfiltered
	return secret, nil
}

```